From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Samuel Mendoza-Jonas <samjonas@amazon.com>
Date: Wed, 24 Jul 2019 17:04:33 -0700
Subject: [PATCH 2/2] shell: Allow root user

For the purposes of Thar-day allow a zero uid/gid when launching a
shell.

Signed-off-by: Samuel Mendoza-Jonas <samjonas@amazon.com>
---
 agent/session/shell/shell_unix.go | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/agent/session/shell/shell_unix.go b/agent/session/shell/shell_unix.go
index 97a542aa..15412277 100644
--- a/agent/session/shell/shell_unix.go
+++ b/agent/session/shell/shell_unix.go
@@ -209,12 +209,8 @@ func getUserCredentials(log log.T, sessionUser string) (uint32, uint32, []uint32
 		groupIds = append(groupIds, uint32(groupIdFromName))
 	}
 
-	// Make sure they are non-zero valid positive ids
-	if uid > 0 && gid > 0 {
-		return uint32(uid), uint32(gid), groupIds, nil
-	}
-
-	return 0, 0, nil, errors.New("invalid uid and gid")
+	// Allow root
+	return uint32(uid), uint32(gid), groupIds, nil
 }
 
 // generateLogData generates a log file with the executed commands.
-- 
2.17.1

